modules/candidate_parameters/js/candidateInfo.js

```diff
@@ -55,7 +55,7 @@ var CandidateInfo = React.createClass({
     formData[formElement] = value;
 
     // Reset 'reason' field
-    if (formElement === "flagged_caveatemptor" && value === "false") {
+    if (formElement === "flaggedCaveatemptor" && value === "false") {
```

Comment:
In fact, it does, but you would also need to change all names and references of selects to be camelCase